Enhance CLI with reusable prompt functions, avoid code repetition, improve repository creation commands, and prepare changes for version 0.3.0

RichardIrala opened the pull request:
Solution for this issue https://github.com/0xsequence/issue-tracker/issues/3311

taylanpince commented on src/utils/index.ts:
```diff
@@ -7,4 +8,21 @@ export function isValidPrivateKey(privateKey: string) {
+export async function promptUserKeyCustomizationDecision () : Promise<boolean> {
+    return await select({
+        message:
+          "Would you like to use test configuration keys for testing, or customize your configuration keys?",
```
Let's change slightly to "Would you like to use default configuration keys for testing..."

RichardIrala replied:
Done. Modified to: "Would you like to use the default configuration keys for testing?"

taylanpince commented on src/marketplace/create_marketplace_boilerplate.ts:
```diff
@@ -12,101 +13,77 @@ export async function createMarketplaceBoilerplate(program: Command, options: an
+                "Please provide the Wallet Type for your project.\nYou can use 'waas' for Embedded Wallet and 'universal' for Universal Wallet.\nFor more information on wallet types: https://docs.sequence.xyz/solutions/wallets/overview",
+            choices: [
+                {
+                    name: "waas",
```
Let's make the name "Embedded Wallet"

RichardIrala replied:
Done

taylanpince commented on src/marketplace/create_marketplace_boilerplate.ts:
```diff
@@ -12,101 +13,77 @@ export async function createMarketplaceBoilerplate(program: Command, options: an
+                    value: "waas",
+                },
+                {
+                    name: "universal",
```
Name should be "Universal Wallet"

RichardIrala replied:
Done

taylanpince commented on src/boilerplates/create_embedded_wallet_react.ts:
```diff
@@ -14,52 +15,56 @@ export async function createEmbeddedWalletReact(program: Command, options: any)
+    const userWantsToConfigureTheirKeys = await promptUserKeyCustomizationDecision();
+    
+    if (userWantsToConfigureTheirKeys) {
+        if (!waasConfigKey) {
```
It feels like we are repeating these `waasConfigKey`, `projectAccessKey` checks and `await input` implementations all over the place. Can you centralize these into a single point as well? That way the implementation will be a lot cleaner.

RichardIrala replied:
About that can you provide me some feedback about this? https://github.com/0xsequence/sequence-cli/pull/11/files#diff-0e4c4ef1bf4917d07d28ca11950aa794bab7500434ff881ba77fee6630501646. I want to know if this satisfies your idea and if it seems well structured to you.

taylanpince replied:
@RichardIrala Very good start. Now let's go one step further and do wrapper utility functions that use `promptForKeyWithLogs` under the hood, but specific to most common repeated cases. I suspect those will be `promptForWaaSConfigKeyWithLogs`, `promptForProjectAccessKeyWithLogs` etc.

RichardIrala replied:
Done

taylanpince reviewed (changes requested):
Minor changes needed, looking great

RichardIrala commented:
# Changes

* Created a new file for types
* Improved some CLI commands by adding missing environment variable prompts
* Added a basic configuration for .prettierrc
* Introduced new functions in the utils folder

## New functions in the utils folder:

* `promptUserKeyCustomizationDecision`
* `promptForKeyWithLogs`
* `promptForProjectAccessKeyWithLogs`
* `promptForWaaSConfigKeyWithLogs`
* `promptForGoogleClientIdWithLogs`
* `promptForAppleClientIdWithLogs`
* `promptForWalletConnectIdWithLogs`
* `writeToEnvFile`
* `writeDefaultKeysToEnvFileIfMissing`

taylanpince reviewed (approved):
This is beautiful – Thank you @RichardIrala
